### ISSUE TITLE
Provide a metric for each loaded certificate in Envoy with information on time left till expiration

### ISSUE THREAD

**Issue by PeterL328** [envoyproxy/envoy#27396]
*Title*: *One line description*
Provide a metric for each loaded certificate in Envoy with information on seconds left till expiration.

*Description*:
>Describe the desired behavior, what scenario it enables and how it
would be used.
Our team is finding the existing stat `server.days_until_first_cert_expiring` and `/certs` endpoint not sufficient for our use case. If we have short-lived certs (couple of hrs) and long-lived public certs, it is not granular enough to capture when our certs are expiring and even if we have a `server.seconds_until_first_cert_expiring`, the short-lived certs expire so fast we would never see the long-lived public certs in this metric until its too late. 

We are proposing adding a new metric for each loaded certificate with a seconds until expiration.
Maybe in the following format:
`sds.<cert_name>.seconds_until_cert_expiring`

I'm happy to work on a PR to implement above but would like to get some suggestions or if someone has alternative ideas?

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

**Comment by PeterL328** [comment]
cc: @sc0ttbeardsley

**Comment by ggreenway** [comment]
That sounds like a reasonable solution.

**Comment by ggreenway** [comment]
I agree that putting it under `sds` is confusing. There are other stats under `listener.<name>.ssl`; that could be a potential location, and also scopes it to a specific listener if the cert doesn't have a name, which helps identify which one it is. You could also put them under `server` so they're next to `server.days_until_first_cert_expiring`.

For no-name certs, if it has a file-path you can use that. For inline-in-yaml, you could number them (`unnamed1`, `unnamed2`, etc).

**Comment by PeterL328** [comment]
Commenting so issue will not be marked stale. Still working on the feature.

**Comment by PeterL328** [comment]
Completed as part of https://github.com/envoyproxy/envoy/pull/27687
